spec/defines/interface_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'spec_helper_puppet'
+
+describe 'jboss::interface', :type => :define do
+  basic_bind_variables_list = [
+    "any-address", "inet-address", "link-local-address",
+    "loopback", "loopback-address", "multicast",
+    "nic", "nic-match", "point-to-point", "public-address",
+    "site-local-address", "subnet-match", "up", "virtual"
+  ]
+  legacy_bind_variables_list = [ "any-ipv4-address", "any-ipv6-address" ]
+
+  let (:generic_facts) do
+    {
+      :operatingsystem => 'OracleLinux',
+      :osfamily        => 'RedHat',
+      :concat_basedir  => '/root/concat',
+      :puppetversion   => Puppet.version,
+    }
+  end
+  let(:title) { 'test-interface' }
+  let(:facts) do
+    generic_facts.merge(runtime_facts)
+  end
+  let(:basic_bind_variables) { Hash[basic_bind_variables_list.map {|x| [x, :undef]}] }
+  let(:legacy_bind_variables) { Hash[legacy_bind_variables_list.map {|x| [x, :undef]}] }
+
+  context 'with jboss_running => true and runasdomain => false parameters set' do
+    let(:runtime_facts) do
+      { :jboss_running => 'true' }
+    end
+    let(:params) do
+      { :runasdomain => 'false' }
+    end
+    let(:pre_condition) { "class { jboss: product => '#{product}', version => '#{version}'}" }
+
+    context 'with product => wildfly and version => 9.0.2.Final parameters set' do
```

Comment:
Test also for not compatible versions. For ex wildfly 15 should `raise_error (Puppet::Error, /Unsupported version Wildfly 15/)`